src/services/push-service.ts

```diff
@@ -156,6 +163,38 @@ export const uploadFileToStorage = async (
   }
 };
 
+const checkFileSizesInDirectory = (directoryPath: string): void => {
+  const checkDirectory = (dir: string, depth = 0) => {
+    if (depth > MAX_RECURSION_DEPTH) {
+      return;
```

Comment:
instead throw a descriptive error for the end user to change the depth of his project to be less than MAX_RECURSION_DEPTH